pytensor/link/pytorch/dispatch/scalar.py

```diff
@@ -0,0 +1,241 @@
+from pytensor.scalar.basic import (
+    Cast,
+    Clip,
+    Identity,
+    ScalarOp,
+    Second,
+)
+from pytensor.scalar.math import (
+    BetaIncInv,
+    Erf,
+    Erfc,
+    Erfcinv,
+    Erfcx,
+    Erfinv,
+    GammaIncCInv,
+    GammaIncInv,
+    Iv,
+    Ive,
+    Log1mexp,
+    Psi,
+    TriGamma,
+)
+
+
+def try_import_tfp_jax_op(op: ScalarOp, jax_op_name: str | None = None) -> Callable:
+    try:
+        import tensorflow_probability.substrates.jax.math as tfp_jax_math
+    except ModuleNotFoundError:
+        raise NotImplementedError(
+            f"No JAX implementation for Op {op.name}. "
+            "Implementation is available if TensorFlow Probability is installed"
+        )
+
+    if jax_op_name is None:
+        jax_op_name = op.name
+    return typing.cast(Callable, getattr(tfp_jax_math, jax_op_name))
+
+
+def all_inputs_are_scalar(node):
+    """Check whether all the inputs of an `Elemwise` are scalar values."""
+    ndims_input = [inp.type.ndim for inp in node.inputs]
+    are_inputs_scalars = True
+    for ndim in ndims_input:
+        try:
+            if ndim > 0:
+                are_inputs_scalars = False
+        except TypeError:
+            are_inputs_scalars = False
+
+    return are_inputs_scalars
+
+
+@pytorch_funcify.register(ScalarOp)
+def pytorch_funcify_ScalarOp(op, node, **kwargs):
+    """Return pytorch function that implements the same computation as the Scalar Op.
+
+    This dispatch is expected to return a pytorch function that works on Array inputs as Elemwise does,
+    even though it's dispatched on the Scalar Op.
+    """
+
+    nfunc_spec = getattr(op, "nfunc_spec", None)
+    if nfunc_spec is None:
+        raise NotImplementedError(f"Dispatch not implemented for Scalar Op {op}")
+
+    func_name = nfunc_spec[0]
+
+    pytorch_func = getattr(torch, func_name)
+
+    if len(node.inputs) > op.nfunc_spec[1]:
+        # Some Scalar Ops accept multiple number of inputs, behaving as a variadic function,
+        # even though the base Op from `func_name` is specified as a binary Op.
+        # This happens with `Add`, which can work as a `Sum` for multiple scalars.
+        pytorch_variadic_func = getattr(torch, op.nfunc_variadic, None)
+        if not pytorch_variadic_func:
+            raise NotImplementedError(
+                f"Dispatch not implemented for Scalar Op {op} with {len(node.inputs)} inputs"
+            )
+
+        def pytorch_func(*args):
+            return pytorch_variadic_func(
+                torch.stack(torch.broadcast_tensors(*args), axis=0), axis=0
+            )
+
+    return pytorch_func
+
+
+@pytorch_funcify.register(Cast)
+def pytorch_funcify_Cast(op, **kwargs):
+    def cast(x):
+        return torch.tensor(x).astype(op.o_type.dtype)
+
+    return cast
+
+
+@pytorch_funcify.register(Identity)
+def pytorch_funcify_Identity(op, **kwargs):
+    def identity(x):
+        return x
+
+    return identity
+
+
+@pytorch_funcify.register(Clip)
```

Comment:
looks like the default scalar op would work for this, no need to implement special dispatch

Reply:
Understood. What about the dispatch for `Composite` op?
Didn't implement it earlier as I wasn't sure how to implement `jnp.vectorize` in pytorch

Reply:
We can live that one for a later PR